src/components/policy/src/policy/include/policy/policy_manager_impl.h

```diff
@@ -187,6 +188,25 @@ class PolicyManagerImpl : public PolicyManager {
 
     virtual std::string RetrieveCertificate() const OVERRIDE;
 
+    class RetrySequenceTimewatch {
+     public:
+      RetrySequenceTimewatch(PolicyManagerImpl* ptr);
+      ~RetrySequenceTimewatch();
+      void Stop();
+      bool IsRunning();
+      void Start(uint32_t timeout);
+
+     private:
+      typedef timer::TimerThread<RetrySequenceTimewatch> Timer;
+      Timer timer_retry_sequence_;
+
+      PolicyManagerImpl* policy_manager_;
+
+     private:
+      RetrySequenceTimewatch();
```

Comment:
@AGritsevich in case you declare anyone constructor for class (with one parameter in your example) you disable all another constructors including constructor without parameters. So no needed to put it to private section